subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveBuildCacheKeyExecuter.java

```diff
@@ -111,8 +110,11 @@ private TaskOutputCachingBuildCacheKey resolve(final TaskInternal task, TaskExec
                 public TaskOutputCachingBuildCacheKey apply(BeforeExecutionState beforeExecutionState) {
                     TaskOutputCachingBuildCacheKey cacheKey = calculator.calculate(task, beforeExecutionState, properties, buildCacheDebugLogging);
                     if (properties.hasDeclaredOutputs() && cacheKey.isValid()) { // A task with no outputs has no cache key.
-                        LogLevel logLevel = buildCacheDebugLogging ? LogLevel.LIFECYCLE : LogLevel.INFO;
-                        LOGGER.log(logLevel, "Build cache key for {} is {}", task, cacheKey.getHashCode());
+                        if (buildCacheDebugLogging) {
+                            LOGGER.warn("Build cache key for {} is {}", task, cacheKey.getHashCode());
```

Comment:
Warn seems like too high. I guess we don't have much choices when we don't want to use the Gradle logger, right?

Reply:
What about `DebuggingTaskOutputCachingBuildCacheKeyBuilder`? This one is using `lifecycle` logging still. Shouldn't the logging level be the same for the debug logging?

Reply:
> Warn seems like too high. I guess we don't have much choices when we don't want to use the Gradle logger, right?

We don't. But in general I wouldn't be too hung upon this, given that the user has no way to distinguish between `WARN` and `LIFECYCLE` events.

> What about `DebuggingTaskOutputCachingBuildCacheKeyBuilder`? This one is using `lifecycle` logging still. Shouldn't the logging level be the same for the debug logging?

That one is handled in https://github.com/gradle/gradle/pull/8873. In any case, for our purposes `LIFECYCLE` == `WARN`.